src/main/java/uk/gov/hmcts/reform/idam/client/IdamClient.java

```diff
@@ -37,17 +45,41 @@
     private IdamApi idamApi;
     private OAuth2Configuration oauth2Configuration;
 
+    private static Cache<String, TokenResponse> accessTokenCache;
+    private final long refreshTokenBeforeExpiry;
+    private final boolean cacheEnabled;
+
     @Autowired
-    public IdamClient(IdamApi idamApi, OAuth2Configuration oauth2Configuration) {
+    public IdamClient(
```

Comment:
Is there any potential problem with changing the constructor signature? I'm thinking about other projects using this library.

Reply:
I don't suppose - it's a regular spring bean/component which will be autoconfigured

Reply:
> Is there any potential problem with changing the constructor signature? I'm thinking about other projects using this library.

that is why default values provided, it should be fine.